### PR TITLE
[HttpClient] fallbackto CURLMOPT_MAXCONNECTS when CURLMOPT_MAX_HOST_CONNECTIONS is not available

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #33462
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/HttpClient/CurlHttpClient.php [hunk 1]
```diff
@@ -72,7 +72,10 @@ public function __construct(array $defaultOptions = [], int $maxHostConnections
             curl_multi_setopt($this->multi->handle, CURLMOPT_PIPELINING, CURLPIPE_MULTIPLEX);
         }
         if (\defined('CURLMOPT_MAX_HOST_CONNECTIONS')) {
-            curl_multi_setopt($this->multi->handle, CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : PHP_INT_MAX);
+            $maxHostConnections = curl_multi_setopt($this->multi->handle, CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : PHP_INT_MAX) ? 0 : $maxHostConnections;
```
Comment (fabpot) [inline src/Symfony/Component/HttpClient/CurlHttpClient.php]: PHP docs says that `curl_multi_setopt()` returns a `bool`.

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/CurlHttpClient.php]: Not sure to get what you mean, did you miss the ternary operation maybe?

Comment (fabpot) [reply src/Symfony/Component/HttpClient/CurlHttpClient.php]: Of course I did :)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**